### PR TITLE
feature(): add ct status when user is offline

### PR DESCRIPTION
**Pull request by chengoconnell**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
- If user is offline -> display `unknown (no internet connection)` label

https://user-images.githubusercontent.com/69536010/185614165-30a61f1b-faf6-4345-9b81-4f113046adab.mov

- If user is disconnected from Prismic -> display `unknown (not connected to Prismic)` label
   - If user attempts to save a custom type while disconnected -> keep status the same and display `unknown (not connected to Prismic)` label.
   

https://user-images.githubusercontent.com/69536010/185616703-4844b22e-a585-433b-98cf-2734ef50ad80.mov

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/slice-machine/components/Badge/StatusBadgeWithTooltip.tsx [hunk 1]
```diff
@@ -23,7 +24,19 @@ const statusEnumToDisplayNameAndTooltip = (status?: string) => {
...
+        statusTooltip:
+          "Data from the remote repository could not be fetched (no internet connection).",
+      };
+    case CustomTypeStatus.UnknownDisconnected:
+      return {
+        statusDisplayName: "Unknown",
+        statusTooltip:
+          "Data from the remote repository could not be fetched (not connected to Prismic).",
+      };
+    case CustomTypeStatus.UnknownDefault:
```
Comment (Heyrows) [inline packages/slice-machine/components/Badge/StatusBadgeWithTooltip.tsx]: i'd remove that case, it seems to be the same as the default case :)

File: packages/slice-machine/components/Badge/StatusBadgeWithTooltip.tsx [hunk 2]
```diff
@@ -45,6 +58,12 @@ interface StatusBadgeWithTooltipProps {
 export const StatusBadgeWithTooltip: React.FC<StatusBadgeWithTooltipProps> = ({
   customType,
 }) => {
+  const isOnline = useNetwork();
+
+  customType.__status = !isOnline
```
Comment (Heyrows) [inline packages/slice-machine/components/Badge/StatusBadgeWithTooltip.tsx]: Create a new object please 🙏 , modifying the existing one could create side effects.

File: packages/slice-machine/lib/builders/CustomTypeBuilder/Layout/Header/primaryButton.tsx [hunk 3]
```diff
@@ -51,7 +51,9 @@ export const PrimaryButton: React.FC<PrimaryButtonProps> = ({
     [
       CustomTypeStatus.New,
       CustomTypeStatus.Modified,
-      CustomTypeStatus.Unknown,
+      CustomTypeStatus.UnknownDefault,
```
Comment (Heyrows) [inline packages/slice-machine/lib/builders/CustomTypeBuilder/Layout/Header/primaryButton.tsx]: judging by the previus code we don't need this one

Comment (Heyrows) [reply packages/slice-machine/lib/builders/CustomTypeBuilder/Layout/Header/primaryButton.tsx]: I think the only condition needs is
```suggestion
     customTypeStatus != CustomTypeStatus.synced
```

File: packages/slice-machine/src/modules/availableCustomTypes/index.ts [hunk 4]
```diff
@@ -177,14 +182,18 @@ export const availableCustomTypesReducer: Reducer<
...
+        newLocalCustomType.__status === CustomTypeStatus.UnknownDisconnected;
 
       const remoteCustomType: CustomTypeSM | undefined = state[id].remote;
 
       const newCustomType = {
         ...state[id],
         local: {
           ...newLocalCustomType,
-          __status: getCustomTypeStatus(newLocalCustomType, remoteCustomType),
+          __status: isCustomTypeDisconnected
```
Comment (Heyrows) [inline packages/slice-machine/src/modules/availableCustomTypes/index.ts]: this is not needed, as Lorenzo mentioned, if we are able to push and it comes here, it means the users is connected.

**Review by sintinilorenzo — APPROVED** [review]

File: packages/slice-machine/src/modules/availableCustomTypes/index.ts [hunk 5]
```diff
@@ -182,18 +182,14 @@ export const availableCustomTypesReducer: Reducer<
       const id = action.payload.customTypeId;
       const newName = action.payload.newCustomTypeName;
       const newLocalCustomType = { ...state[id].local, label: newName };
-      const isCustomTypeDisconnected =
-        newLocalCustomType.__status === CustomTypeStatus.UnknownDisconnected;
```
Comment (Heyrows) [inline packages/slice-machine/src/modules/availableCustomTypes/index.ts]: now that I think about it, we should do that if it's any Unknown.

Comment (Heyrows) [reply packages/slice-machine/src/modules/availableCustomTypes/index.ts]: Not for the push, but for the Save.
If the user is not connected to Prismic it wuld also write "New" or maybe i'm wrong and we already handle that case.

**Review by Heyrows — APPROVED** [review]

**Pull request merged**